hera_sim/defaults.py

```diff
@@ -112,24 +116,43 @@ def deactivate(self):
         """Revert to function defaults."""
         self._use_season_defaults = False
 
-    def _get_config(self, config_file):
+    def _get_config(self, config):
         """Validate the choice of configuration file."""
-        assert isinstance(config_file, str), \
-                "Default configurations are set by passing a path to a " \
-                "configuration file or one of the season keys. The " \
-                "currently supported season configurations are " \
-                "{}.".format(SEASON_CONFIGS.keys())
-
-        if config_file in SEASON_CONFIGS.keys():
-            return SEASON_CONFIGS[config_file]
+        #assert isinstance(config, str), \
+        #        "Default configurations are set by passing a path to a " \
+        #        "configuration file or one of the season keys. The " \
+        #        "currently supported season configurations are " \
+        #        "{}.".format(SEASON_CONFIGS.keys())
+        if isinstance(config, str):
+            if config in SEASON_CONFIGS.keys():
+                config = SEASON_CONFIGS[config]
+                # return SEASON_CONFIGS[config]
+            with open(config, 'r') as conf:
+                defaults = yaml.load(conf.read(), Loader=yaml.FullLoader)
+            return defaults
+        elif isinstance(config, dict):
+            return config
         else:
-            return config_file
+            raise ValueError(
+                    "The configuration must be a dictionary or an absolute " \
+                    "path to a configuration YAML." )
+        #else:
+        #    return config
 
     def _check_config(self):
         """Confirm that the specified configuration file can be found."""
-        assert path.exists(self._config), \
-                "Please ensure that a path to the specified configuration " \
-                "file exists. {} could not be found".format(self._config)
+        #assert path.exists(self._config), \
+        #        "Please ensure that a path to the specified configuration " \
+        #        "file exists. {} could not be found".format(self._config)
+        error_message = "Your configuration should be formatted as a nested " \
+                "dictionary: {module: {model: {params: values}}}"
+        try:
+            module = list(self._config.keys())[0]
+            model = list(self._config[module].keys())[0]
+        except KeyError:
+            raise AssertionError(error_message)
+        # seems a little stupid to use self._config rather than self()
```

Comment:
so why don't you just use `self(module, model)`?

Reply:
Done.